src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -424,6 +419,13 @@ internal static bool IsClosureScope(SyntaxNode node)
                     return true;
 
                 default:
+                    // With the introduction of pattern-matching, many nodes now contain top-level
+                    // expressions that may introduce pattern variables.
+                    if (node.Parent.IsKind(SyntaxKind.EqualsValueClause))
+                    {
+                        return true;
+                    }
```

Comment:
A local variable declaration contains an EqualsValueClause, but its expression is not a closure scope, so judging by the name of this method this might not be right. I don't understand how this method is used so I can't tell if it is OK or not.

Reply:
Given the doc comment, the function doesn't have to be 100% accurate and it was not ruling out EqualsValueClause from local declarations before.
```
        /// <summary>
        /// Returns true if the specified node is of a kind that could represent a closure scope -- that
        /// is, a scope of a captured variable.
        /// Doesn't check whether or not the node actually declares any captured variable.
        /// </summary>
```

---
In reply to: [174358539](https://github.com/dotnet/roslyn/pull/25441#discussion_r174358539) [](ancestors = 174358539)

Reply:
> the function doesn't have to be 100% accurate

I wouldn't say that. It has to accurately identify syntax that may be a closure scope.

---
In reply to: [174535499](https://github.com/dotnet/roslyn/pull/25441#discussion_r174535499) [](ancestors = 174535499,174358539)

Reply:
> I wouldn't say that. It has to accurately identify syntax that may be a closure scope.

That is what I meant. It doesn't have to be 100% accurate in ruling out nodes

---
In reply to: [174560439](https://github.com/dotnet/roslyn/pull/25441#discussion_r174560439) [](ancestors = 174560439,174535499,174358539)